nav2_util/include/nav2_util/validate_messages.hpp

```diff
@@ -55,6 +55,17 @@ bool validateMsg(const double & num)
   return true;
 }
 
+template<size_t N>
+bool validateMsg(const std::array<double, N> & msg)
```

Comment:
Vector?

Reply:
I noticed `YAML::List` is always transformed into array in ROS, which is defined in `geometry_msg`:

```cpp
  using _pose_type =
    geometry_msgs::msg::Pose_<ContainerAllocator>;
  _pose_type pose;
  using _covariance_type =
    std::array<double, 36>;
  _covariance_type covariance;
```

Reply:
Ah got it, this is for the covariance checking :+1: Can you add a comment for this one that this is what this is for? I'll forget in a year and when I look over this again I'll wonder again why its not a vector :laughing:

Reply:
of course 😆